python_frank_energie/models.py

```
@@ -39,44 +51,300 @@ def from_dict(data: dict[str, str]) -> Authentication:
+        )
+
 
 @dataclass
 class Invoices:
-    """Invoices data, including the previous, current and upcoming period."""
+    """Represents invoices information, including the previous, current, and upcoming period."""
+
+    def __init__(
+        self,
+        allPeriodsInvoices: Optional[list[Invoice]
+                                     ] = field(default_factory=list),
```

Ruff

Do not perform function call `field` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)